BusinessLayer/ReverseLineReader.cs

```diff
@@ -0,0 +1,292 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace BusinessLayer
+{
+
+    /// <summary>
+    /// Takes an encoding (defaulting to UTF-8) and a function which produces a seekable stream
+    /// (or a filename for convenience) and yields lines from the end of the stream backwards.
+    /// Only single byte encodings, and UTF-8 and Unicode, are supported. The stream
+    /// returned by the function must be seekable.
+    /// </summary>
+    public sealed class ReverseLineReader : IEnumerable<string>
```

Comment:
this copy pasted file is ugly, remove all useless code and refactor what is remaining + add a comment to explain where you found it

Reply:
Add link to original file.
Don't think there is any useless code in it (maybe remove comments ?)